File: runner.js
var recursion1 = require('./recursion-1.js');
var recursion3 = require('./recursion-3.js');

var example = process.argv[2];
var n = parseInt(process.argv[3]);

switch(example) {
  case "factorial":
    console.log(recursion1.factorial(n));
    break;
  case "count-up":
    console.log(recursion3.countUp(n));
    break;
}

File: runner.clj
(ns runner)

(load-file "./recursion-1.clj")
(load-file "./recursion-2.clj")
(load-file "./recursion-3.clj")
(load-file "./recursion-4.clj")
(load-file "./recursion-5.clj")

(defn run
  [[name n]]
   (case (keyword name)
     :factorial (recursion-1/factorial n)
     :factorial-step (recursion-2/factorial n)
     :count-up (recursion-3/count-up n)
     :count-up-tail (recursion-4/count-up n)
     :count-up-recur (recursion-5/count-up n)))

(println (run (map read-string (rest *command-line-args*))))
